lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2303,22 +2303,17 @@ func (d *zfs) readonlySnapshot(vol Volume) (string, revert.Hook, error) {
 	d.logger.Debug("Created backup snapshot", logger.Ctx{"dev": srcSnapshot})
 
 	if d.isBlockBacked(vol) {
-		// For block devices, we make them appear.
-		// Check if already active.
-		current, err := d.getDatasetProperty(dataset, "volmode")
+		snapdevMode, err := d.getDatasetProperty(dataset, "snapdev")
 		if err != nil {
 			return "", nil, err
 		}
 
-		if current != "dev" {
-			// Activate.
-			err = d.setDatasetProperties(dataset, "volmode=dev", "snapdev=visible")
+		if snapdevMode != "visible" {
+			err = d.setDatasetProperties(dataset, "snapdev=visible")
 			if err != nil {
 				return "", nil, err
 			}
 
-			defer func() { _ = d.setDatasetProperties(dataset, "volmode=none", "snapdev=hidden") }()
```

Comment:
Should this defer not still be here to undo the changes ?

Reply:
Fixed.